src/ttsim/convert_nested_data.py

```diff
@@ -3,17 +3,49 @@
 from typing import TYPE_CHECKING
 
 import dags.tree as dt
+import numpy as np
 import optree
 import pandas as pd
 
 from ttsim.shared import format_errors_and_warnings, format_list_linewise
 
 if TYPE_CHECKING:
-    from ttsim.typing import NestedData, NestedInputsPathsToDfColumns
+    from ttsim.typing import NestedData, NestedStrings, QualNameData
 
 
-def create_data_tree_from_df(
-    inputs_tree_to_df_columns: NestedInputsPathsToDfColumns,
+def nested_data_to_dataframe(
+    nested_data: NestedData,
+    targets_tree_to_outputs_df_columns: NestedStrings,
+    index_column: pd.Series,
```

Comment:
```suggestion
    data_with_p_id: NestedData,
```
Thinking ahead here, I think we should just require a dataset that has allows us to index with `p_id`.